### ISSUE TITLE
Installing on Dockers' Python 3.7 container fails

### ISSUE THREAD

**Issue by Carelvd** [mkleehammer/pyodbc#644]
Please first make sure you have looked at:

I have tried to go through the available Issues and Stack Overflow Questions but have not had any luck with an answer

### Environment

To diagnose, we usually need to know the following, including version numbers.  On Windows, be
sure to specify 32-bit Python or 64-bit:

- Python: 3.7
- pyodbc: 4.0.27
- OS: Python 3.7:latest docker container
- DB: Microsoft SQL (Also in the mcr.microsoft.com/server/mssql:ubuntu container; not sure this is relevant though)
- driver: Not Sure

### Issue

I was trying to pull down the PyODBC package while setting up a flask environment. I was hoping the package would install correctly but it fails with the following error message. I was hoping to be more helpful in this message but I'm not familiar with any of the components reported in the error.

    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-af6iayzo/pyodbc/setup.py'"'"'; __file__='"'"'/tmp/pip-install-af6iayzo/pyodbc/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-zdwz9cwb/install-record.txt --single-version-externally-managed --compile
         cwd: /tmp/pip-install-af6iayzo/pyodbc/
    Complete output (14 lines):
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYODBC_VERSION=4.0.27 -I/usr/local/include/python3.7m -c src/buffer.cpp -o build/temp.linux-x86_64-3.7/src/buffer.o -Wno-write-strings
    In file included from src/buffer.cpp:12:
    src/pyodbc.h:56:10: fatal error: sql.h: No such file or directory
     #include <sql.h>
              ^~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
    ERROR: Command errored out with exit status 1: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-af6iayzo/pyodbc/setup.py'"'"'; __file__='"'"'/tmp/pip-install-af6iayzo/pyodbc/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-zdwz9cwb/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.

It seems that the `sql.h` file is missing, this probably this is due to me installing the package in a  container and the environment not having all of necessary the header files, 

It seems that one should use the prebuilt binaries by Christophe Gholke as per issue #386. Issue #165 appears related.

**Comment by v-makouz** [comment]
You need a driver manager. I'm guessing this is Ubuntu? If so run `sudo apt-get install unixodbc-dev`
(on RedHat it would be `sudo yum install unixODBC-devel`)

**Comment by Carelvd** [comment]
@v-makouz Thank you for the response, I tried installing the driver at the time only to find I had a number of other unrelated issues e.g. the container I was using required 2G of RAM and my whole setup runs on a VM with 2G RAM. After fiddling about for a while I ended up porting the PyODBC code to the PyMsSQL driver and things are now working. I will revisit PyODBC in the near future and report back but for now I'm marking this as closed.
